components/ws-daemon/pkg/cpulimit/cfs.go

```diff
@@ -60,31 +57,66 @@ func (basePath CgroupCFSController) SetLimit(limit Bandwidth) (changed bool, err
 
 func (basePath CgroupCFSController) readParentQuota() time.Duration {
 	parent := CgroupCFSController(filepath.Dir(string(basePath)))
-	pq, err := parent.readUint64("cpu.cfs_quota_us")
+	pq, err := parent.readCfsQuota()
 	if err != nil {
 		return time.Duration(0)
 	}
 
 	return time.Duration(pq) * time.Microsecond
 }
 
-func (basePath CgroupCFSController) readUint64(path string) (uint64, error) {
+func (basePath CgroupCFSController) readString(path string) (string, error) {
 	fn := filepath.Join(string(basePath), path)
 	fc, err := os.ReadFile(fn)
 	if err != nil {
-		return 0, err
+		return "", err
 	}
 
 	s := strings.TrimSpace(string(fc))
-	if s == "max" {
-		return math.MaxUint64, nil
+	return s, nil
+}
+
+func (basePath CgroupCFSController) readCfsPeriod() (time.Duration, error) {
+	s, err := basePath.readString("cpu.cfs_period_us")
+	if err != nil {
+		return 0, err
+	}
+
+	p, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return time.Duration(uint(p)) * time.Microsecond, nil
+}
+
+func (basePath CgroupCFSController) readCfsQuota() (time.Duration, error) {
+	s, err := basePath.readString("cpu.cfs_quota_us")
+	if err != nil {
+		return 0, err
+	}
+
+	p, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	if p < 0 {
+		return time.Duration(math.MaxInt64), nil
+	}
+	return time.Duration(p) * time.Microsecond, nil
+}
+
+func (basePath CgroupCFSController) readCpuUsage() (time.Duration, error) {
+	s, err := basePath.readString("cpuacct.usage")
+	if err != nil {
+		return 0, err
 	}
 
 	p, err := strconv.ParseInt(s, 10, 64)
 	if err != nil {
 		return 0, err
 	}
-	return uint64(p), nil
```

Comment:
When `p` equals 1, `uint64(p)` was an overflow, and it was not an error 😭

Reply:
Hm. But it doesn't seen like you are handling `-1` below? You still doing `uint64(p)` conversion below.
Though I am not sure if `cpuacct.usage` can ever have a negative value?

Reply:
@sagor999 Oh, I misplaced my comment. It was `cfs_quota_us` that could cause an overflow 😂 
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/resource_management_guide/sec-cpu